File: MATLAB/pe094.m


clc; clear;

  i = -1;
  n = 1;
Sum = 0;

while(true)

x = -1/2 * (-(2 - sqrt(3))^n - (2 + sqrt(3))^n);

a = (2*x + i)/3;
b = a;
c = a + i;
P = a + b + c;

if(P > 1000000000); break; end

if(a >= 1 && b >= 1 && c >= 1 )
Sum = Sum + P;
end

i = i * -1;
n = n + 1;


end

Sum 

File: C++/pe094.cpp
#include <iostream>
#include <cmath>
#include <gmpxx.h>
#include <iomanip>

int main() {


mpz_class N = 500000000, Sum = 0;
mpf_class a, b, c, A, perimeter;

for (mpz_class i = 1; i < N; i++)
{
    a = i;
    b = i; 
    c = i + 1;

    A = sqrt((a + b + c)*(a + b - c)*(b + c - a)*(c + a - b)/16.) * 1.0;

    perimeter = a + b + c;

    if(A == static_cast<mpz_class>(A) && A > 0 && perimeter <= N) Sum = Sum + perimeter;
     
    c = i - 1;
    A = sqrt((a + b + c)*(a + b - c)*(b + c - a)*(c + a - b)/16.) * 1.0;

    perimeter = a + b + c;

    if(A == static_cast<mpz_class>(A) && A > 0 && perimeter <= N) Sum = Sum + perimeter;
    
    if(perimeter > N) break;

}
std::cout << std::fixed << std::setprecision(0);
std::cout << Sum << std::endl;

}
